### PR TITLE
Closes #264, #288 cleanup assertions and continue deprecation process

### LINKED ISSUE #264: Feature Request: Clean up assertions
### Feature Idea

The assertion functions should be reviewed and duplicates and functions which are no longer used should be removed. For example `assert_function()` covers `assert_function_params()`.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/assertions.R [hunk 1]
```diff
@@ -871,41 +869,35 @@ assert_named <- function(arg, optional = FALSE) {
...
-      "`%s` must be a named list of expressions created using `rlang::exprs()` but is %s",
-      arg_name(substitute(arg)),
-      what_is_it(arg)
-    )
-    abort(err_msg)
-  }
-
-  invisible(arg)
+  deprecate_warn("0.6.0", "assert_named_exprs()", "assert_expr_list()", always = TRUE)
+  assert_expr_list(arg = arg, named = TRUE)
```
Comment (bms63) [inline R/assertions.R:889]: Should this be 0.5 as our cran release is 0.4?

File: R/assertions.R [hunk 2]
```diff
@@ -915,8 +907,8 @@ assert_named_exprs <- function(arg, optional = FALSE) {
 #'
 #' @export
 #'
-#' @keywords assertion
-#' @family assertion
+#' @keywords deprecated
+#' @family deprecated
```
Comment (bms63) [inline R/assertions.R:911]: Do you mind deleting the examples to follow our deprecation strategy?

image

**Review by bms63 — CHANGES_REQUESTED** [review]
I think we just need to remove the examples and check version numbers and we are good to go.

**Review by bms63 — APPROVED** [review]
Looks good to me!
